File: Packages/Flip54Core/Sources/Flip54Core/Suit.swift
public enum Suit: String, Codable, CaseIterable, Hashable, Sendable {
    case hearts, spades, clubs, diamonds

    public var color: SuitColor {
        switch self {
        case .hearts, .diamonds: .red
        case .spades, .clubs:   .black
        }
    }

    public var bodyFocusLabel: String {
        switch self {
        case .hearts:   return "Lower Body"
        case .spades:   return "Upper Body"
        case .clubs:    return "Total Body"
        case .diamonds: return "Core"
        }
    }

    public var suitCharacter: String {
        switch self {
        case .hearts:   return "♥"
        case .spades:   return "♠"
        case .clubs:    return "♣"
        case .diamonds: return "♦"
        }
    }
}

public enum SuitColor: String, Codable, Sendable {
    case red, black
}

File: android/core/src/main/kotlin/com/flip54/core/Suit.kt
package com.flip54.core

import kotlinx.serialization.Serializable

@Serializable
enum class Suit {
    HEARTS, SPADES, CLUBS, DIAMONDS;

    val color: SuitColor get() = when (this) {
        HEARTS, DIAMONDS -> SuitColor.RED
        SPADES, CLUBS    -> SuitColor.BLACK
    }
}

@Serializable
enum class SuitColor { RED, BLACK }
